filebeat/module/haproxy/log/_meta/fields.yml

```diff
@@ -29,16 +26,44 @@
     fields:
     - name: captured_cookie
       description: >
-        captured_cookie is an optional "name=value" entry indicating that the server has returned a cookie with its request.
+        Optional "name=value" entry indicating that the server has returned a cookie with its request.
 
     - name: captured_headers
       description: >
-        captured_request_headers is a list of headers captured in the request due to the presence of the "capture request header" statement in the frontend.
+        List of headers captured in the request due to the presence of the "capture request header" statement in the frontend.
       type: text
 
     - name: raw_request_line
-      description: raw_request_line is the complete HTTP request line, including the method, request and HTTP version string.
+      description: Complete HTTP request line, including the method, request and HTTP version string.
       type: text
-  
+      
+    - name: time_active_ms
+      description: Time the request remained active in haproxy, which is the total time in milliseconds elapsed between the first byte of the request was received and the last byte of response was sent.
+      type: long
 
+    - name: time_wait_without_data_ms
+      description: Total time in milliseconds spent waiting for the server to send a full HTTP response, not counting data.
+      type: long
 
+    - name: time_wait_ms
+      description: Total time in milliseconds spent waiting for a full HTTP request from the client (not counting body) after the first byte was received.
+      type: long
+  
+- name: default
+  description: Default HAProxy log format https://cbonte.github.io/haproxy-dconv/1.7/configuration.html#8.2.1
+  type: group
+  fields:
+  - name: mode
+    type: text
+    description: mode that the frontend is operating (TCP or HTTP)
```

Comment:
I think this can also be common to multiple logs, remove it too from the default namespace, and it can be set to `TCP` on the tcp log format.
A possible name for this could be `haproxy.protocol`, or `network.protocol` as of ECS.

Reply:
Let's not use `network.protocol` yet from ECS as it still might change.

Reply:
As far as I can read in the HAProxy documentation,`mode` is only specific for _Default_ format https://cbonte.github.io/haproxy-dconv/1.7/configuration.html#8.2.1 In TCP format there is no "mode" at all (mode is always TCP).

About the naming, I thought that we already agreed that we were going to leave ECS apart in this PR and focus on maintain naming of the service, which is the most familiar naming for an HAProxy user who could use this module.

Reply:
Yes, `mode` seems to be only specified in the default format, it wouldn't make sense in protocol-specific formats, but it can still be useful in events so users can differentiate metrics from tcp and http connections.

We can set `haproxy.protocol` to the value of mode for the default formats, and for protocol-specific formats set it to the specific protocol depending on the matching pattern. I'm not sure how to set a field depending on the matching pattern, we can leave it for a future change, but I'd use a common name like `haproxy.protocol` instead of `haproxy.default.mode` in any case, thinking in the future uses of this field.

Reply:
I think we are spending too much time and energy in a field that only appears (and gets parsed) in a deprecated log format.

I feel like writing complex conditional parsing for non-straightforward (invented) fields which doesn't usually appear in HAProxy and to get something that we haven't measure and that it has not come from a user requirements is over-engineering things a bit

WDYT?

Reply:
Yes, as I said we can leave adding this field in other formats for future changes. I was just asking about using a generic name like `haproxy.protocol` (instead of `haproxy.default.mode`), no overengineering, just using a more future-proof name, wdyt about this?

Reply:
Or maybe `haproxy.mode` if you want to keep haproxy naming, but without the `default` namespacing.